### ISSUE TITLE
Switch ranked ballot representation?

### ISSUE THREAD

**Issue by endolith** [endolith/elsim#8]
Currently it's using the representation that matches how we often write ballots:

```
A > B > C
B > C > A
B > A > C
```

becomes

```
0, 1, 2
1, 2, 0
1, 0, 2
```

Where rows are voters, columns are rankings, and cells contain candidate IDs.

But this format doesn't allow for expressions of indifference, and for elimination methods, this requires either shuffling cells around or separate pointers for each row and is a bit clunky.

The other convention is more general because it allows indifference:

```
1, 2, 3
3, 1, 2
2, 1, 3
```

Here the rows are voters, columns are candidates, and cells store rankings.  Does this also simplify or speed up the calculation of elimination methods?

**Comment by endolith** [comment]
DeepSeek:

I’ve looked into this a fair bit, and here’s where I’ve landed:

- The current representation (rows=voters, columns=preference positions) is already optimal for IRV — it gives O(1) access to each voter’s current top choice and keeps the whole algorithm at O(V·C) with no per-round overhead.
- The rank matrix (rows=voters, columns=candidates, cells=rank) is great for methods like Borda (just sum a column) and it naturally supports equal rankings. However, for IRV it doesn’t actually speed anything up — you’d still need to convert each voter’s ranks into an ordered list to get the same performance, otherwise you’re scanning all candidates every round.
- Since IRV itself doesn’t allow indifference, the equal‑ranking capability of the rank matrix doesn’t benefit IRV directly.

So my suggestion would be to **keep the existing ballot format for IRV and other strict‑ranking elimination methods** (where it shines), and add a small utility that converts a rank‑matrix ballot set into the current format (an `order` array per voter) — that conversion is O(V·C) and fast. This way users who have rank‑matrix data can easily feed it into IRV, and we leave the door open for future methods that do support equal rankings to use the rank matrix natively, without touching the IRV hot path.

In other words: don’t switch the core representation, just bridge the two formats with a conversion step.
